lib/posix/clock.c

```diff
@@ -68,12 +91,22 @@ int clock_gettime(clockid_t clock_id, struct timespec *ts)
 int clock_settime(clockid_t clock_id, const struct timespec *tp)
 {
 	struct timespec base;
+#ifdef CONFIG_RTC
+	struct device *dev_rtc;
+#endif
 
 	if (clock_id != CLOCK_REALTIME) {
 		errno = EINVAL;
 		return -1;
 	}
 
+#ifdef CONFIG_RTC
```

Comment:
you could consider using `IS_ENABLED` macro instead of ifdefs.

Reply:
With ifdef, the code unneeded is removed..

Reply:
same with IS_ENABLED() but then compiler removes it and not preprocessor. I has advantage that code is compiled even though there is no sample/test which enables given feature - no dead code.

Reply:
Ok, it's a good reason!